File: server/src/main/java/bit/project/server/entity/Broker.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;


@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Broker {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDateTime tocreation;

    @Lob
    private String description;

    private String name;

    private String contact1;

    private String contact2;

    private String nic;

    private String passport;

    @Lob
    private String address;

    private String email;

    private String fax;


    @ManyToOne
    private Nametitle nametitle;

    @JsonIgnore
    @OneToMany(mappedBy = "broker")
    private List<Offerrequest> brokerList;

    @ManyToOne
    @JsonIgnoreProperties({"creator","tocreation","roleList"})
    private User creator;


    public Broker(Integer id){
        this.id = id;
    }




    public Broker(Integer id, String code, String name) {
        this.id = id;
        this.code = code;
        this.name = name;

    }


}

File: client/src/app/entities/broker.ts
import {User} from './user';
import {DataPage} from '../shared/data-page';
import {Nametitle} from './nametitle';

export class Broker {
  id: number;
  code: string;
  tocreation: string;
  description: string;
  name: string;
  contact1: string;
  contact2: string;
  nic: string;
  passport: string;
  address: string;
  email: string;
  fax: string;
  creator: User;
  nametitle: Nametitle;




  constructor(id: number = null) {
    this.id = id;
  }
}


export class BrokerDataPage extends DataPage {
  content: Broker[];
}
